src/robin_dict.jl

```diff
@@ -0,0 +1,679 @@
+RobinDict{String,Int64} with 2 entries:
+  "B" => 2
+  "A" => 1
+```
+"""
+mutable struct RobinDict{K,V} <: AbstractDict{K,V}
+    #there is no need to maintain an table_size as an additional variable
+    slots::Array{UInt8,1} # indicator, to be used later on
+    keys::Array{K,1}
+    vals::Array{V,1}
+    dibs::Array{Int,1} # distance to initial bucket - critical for implementation
+    count::Int
+    totalcost::Int
+    maxprobe::Int    # length of longest probe
+    idxfloor::Int
+    max_lf :: Float32
+
+    function RobinDict{K, V}() where {K, V}
+        n = 16 # default size of an empty Dict in Julia
+        new(zeros(UInt, n), Vector{K}(undef, n), Vector{V}(undef, n), zeros(Int, n), 0, 0, 0, 0, 0)
+    end
+
+    function RobinDict{K, V}(d::RobinDict{K, V}) where {K, V}
+        new(copy(d.slots), copy(d.keys), copy(d.vals), copy(d.dibs), d.count, d.totalcost, d.maxprobe, d.idxfloor, d.max_lf)
+    end
+
+    function RobinDict{K, V}(slots, keys, vals, dibs, count, totalcost, maxprobe, idxfloor, max_lf) where {K, V}
+        new(slots, keys, dibs, vals, count, totalcost, maxprobe, idxfloor, max_lf)
+    end
+end
+
+function RobinDict{K,V}(kv) where {K, V}
+    h = RobinDict{K,V}()
+    for (k,v) in kv
+        h[k] = v
+    end
+    h
+end
+RobinDict{K,V}(p::Pair) where {K,V} = setindex!(RobinDict{K,V}(), p.second, p.first)
+function RobinDict{K,V}(ps::Pair{K, V}...) where {K, V}
+    h = RobinDict{K,V}()
+    sizehint!(h, length(ps))
+    for p in ps
+        h[p.first] = p.second
+    end
+    return h
+end
+
+RobinDict() = RobinDict{Any,Any}()
+RobinDict(kv::Tuple{}) = RobinDict()
+
+RobinDict(kv::Tuple{Vararg{Pair{K,V}}}) where {K,V}       = RobinDict{K,V}(kv)
+RobinDict(kv::Tuple{Vararg{Pair{K}}}) where {K}           = RobinDict{K,Any}(kv)
+RobinDict(kv::Tuple{Vararg{Pair{K,V} where K}}) where {V} = RobinDict{Any,V}(kv)
+RobinDict(kv::Tuple{Vararg{Pair}})                        = RobinDict{Any,Any}(kv)
+
+RobinDict(kv::AbstractArray{Tuple{K,V}}) where {K,V} = RobinDict{K,V}(kv)
+RobinDict(kv::AbstractArray{Pair{K,V}}) where {K,V}  = RobinDict{K,V}(kv)
+RobinDict(kv::AbstractDict{K,V}) where {K,V} = RobinDict{K,V}(kv)
+
+RobinDict(ps::Pair{K,V}...) where {K,V} = RobinDict{K,V}(ps)
+RobinDict(ps::Pair{K}...,) where {K}             = RobinDict{K,Any}(ps)
+RobinDict(ps::(Pair{K,V} where K)...,) where {V} = RobinDict{Any,V}(ps)
+RobinDict(ps::Pair...) = RobinDict{Any,Any}(ps)
+
+function RobinDict(kv)
+    try
+        dict_with_eltype(kv, eltype(kv))
+    catch e
+        if isempty(methods(iterate, (typeof(kv),))) ||
+            !all(x->isa(x,Union{Tuple,Pair}),kv)
+            throw(ArgumentError("RobinDict(kv): kv needs to be an iterator of tuples or pairs"))
+        else
+            rethrow(e)
+        end
+    end
+end
+
+dict_with_eltype(kv, ::Type{Tuple{K,V}}) where {K,V} = RobinDict{K,V}(kv)
+dict_with_eltype(kv, ::Type{Pair{K,V}}) where {K,V} = RobinDict{K,V}(kv)
+dict_with_eltype(kv, t) = RobinDict{Any,Any}(kv)
```

Comment:
These seem to be from an old implementation of `dict_with_eltype` and should be dropped